resources/queue.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	nitricv1 "github.com/nitrictech/apis/go/nitric/v1"
+	"github.com/nitrictech/go-sdk/api/queues"
+)
+
+type QueuePermission string
+
+const (
+	QueueSending  QueuePermission = "sending"
+	QueueReceving QueuePermission = "receiving"
+)
+
+var (
+	QueueEverything []QueuePermission = []QueuePermission{QueueSending, QueueReceving}
+)
+
+func NewQueue(name string, permissions ...QueuePermission) (queues.Queue, error) {
+	return run.NewQueue(name, permissions...)
+}
+
+func (m *manager) NewQueue(name string, permissions ...QueuePermission) (queues.Queue, error) {
```

Comment:
Not a big deal but it might be worth considering caching named references for resource types so we can skip duplicate re-declaration (this should be handled by the server side so its more of an optimization that could be skipped or done later).